api/filter/http_connection_manager.proto

```diff
@@ -217,6 +217,50 @@ message HttpConnectionManager {
     // used to populate the tag value. The tag is created if the specified
     // header name is present in the request’s headers.
     repeated string request_headers_for_tags = 2;
+
+    // Criteria for identifying a request to be decorated
+    message DecoratorMatch {
+      // A path specifier must be present.
+      oneof path_specifier {
+        // If specified, this is a prefix rule meaning that the prefix must
+        // match the beginning of the :path header.
+        string prefix = 1;
```

Comment:
I like the idea, but does this need to be independent to the existing https://github.com/lyft/envoy-api/blob/master/api/rds.proto#L67 RouteMatch message? Can we attach a `Decorator` to a `Route` https://github.com/lyft/envoy-api/blob/master/api/rds.proto#L288?

Reply:
So for example?
```
                  "routes": [
                    {
                      "timeout_ms": 0,
                      "prefix": "/trace",
                      "cluster": "service2",
                      "method": "GET",
                      "operation": "getTrace"
                    }
                  ]
```
Then the `method` would become part of the routing criteria and potentially a route would need to be defined for each REST endpoint.
If that is the preference, then I could look at changing - but would be good to get opinions from others?

Reply:
If the user is going to have to basically build a route table inside the tracing config, why not just build a more specific route table as you demonstrate above, and just allow specifying an operation name on the route. (This could be used in the future for stats, logging, and other things potentially).

That seems better to me and more future proof. Any reason not to do that?

If we don't want to do that, I think we could probably move RouteMatch message out and use it in both places. (Note that you don't need to have a method specific match. You can do a header match on the `:method` header).

Reply:
Ok will rework to add to `Route` - will close this PR and reopen with new version.